test/conformance/ingressv2/websocket.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/gorilla/websocket"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/net-ingressv2/test"
+	gwv1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
+)
+
+// TestWebsocket verifies that websockets may be used via a simple Ingress.
+func TestWebsocket(t *testing.T) {
+	t.Parallel()
+	ctx, clients := context.Background(), test.Setup(t)
+
+	const suffix = "- pong"
+	name, port, _ := CreateWebsocketService(ctx, t, clients, suffix)
+
+	domain := name + ".example.com"
+
+	// Create a simple Ingress over the Service.
+	_, dialCtx, _ := createHTTPRouteReadyDialContext(ctx, t, clients, gwv1alpha1.HTTPRouteSpec{
+		Gateways:  testGateway,
+		Hostnames: []gwv1alpha1.Hostname{gwv1alpha1.Hostname(domain)},
+		Rules: []gwv1alpha1.HTTPRouteRule{{
+			ForwardTo: []gwv1alpha1.HTTPRouteForwardTo{{
+				Port:        &port,
+				ServiceName: &name,
+			}},
+		}}})
+
+	dialer := websocket.Dialer{
+		NetDialContext:   dialCtx,
+		Proxy:            http.ProxyFromEnvironment,
+		HandshakeTimeout: 45 * time.Second,
+	}
+
+	u := url.URL{Scheme: "ws", Host: domain, Path: "/"}
+
+	// TODO: As Ingress v2 does not have prober, it needs to make sure backend is ready.
+	waitForwebsocketBackend(t, &dialer, &u)
+
+	conn, _, err := dialer.Dial(u.String(), http.Header{"Host": {domain}})
+	if err != nil {
+		t.Fatal("Dial() =", err)
+	}
+	defer conn.Close()
+
+	for i := 0; i < 100; i++ {
```

Comment:
Why do we need to repeat 100 times?

Reply:
`checkWebsocketRoundTrip()` tries to send 100 random messages "`ping - $RANDOM_NUM`" for one connection and verifies that we can receive the random messages respectively. I think it is something like websocket's functionality test.

The test exists since initial websocket test https://github.com/knative/serving/pull/6355 and I think we can keep it as it is.